src/models/github-server-app.test.ts

```diff
@@ -128,6 +135,92 @@ describe("GitHubServerApp", () => {
+				const uuid = "c97806fc-c433-4ad5-b569-bf5191590be2";
+				const clientIdWithTypo = "lvl.1n23j12389wndd";
+				const originalWebhookSecret = "anothersecret";
+				const originalPrivateKey = "privatekey";
+				let gitHubServerApp;
+
+				beforeEach(async () => {
+					gitHubServerApp = await GitHubServerApp.create({
+						uuid,
+						appId: 1,
+						gitHubAppName: "my awesome app",
+						gitHubBaseUrl,
+						gitHubClientId: clientIdWithTypo,
+						gitHubClientSecret: "secret",
+						webhookSecret: "anothersecret",
+						privateKey: "privatekey",
+						installationId
+					});
+				});
+
+				it("should update GitHub app when uuid is found", async () => {
+					const originalApp = await GitHubServerApp.findForUuid(uuid);
+					expect(originalApp && originalApp.gitHubClientId).toEqual(clientIdWithTypo);
+					expect(originalApp && originalApp.webhookSecret).toEqual(`encrypted:${originalWebhookSecret}`);
+					expect(originalApp && originalApp.privateKey).toEqual(`encrypted:${originalPrivateKey}`);
+
+					await GitHubServerApp.updateGitHubAppByUUID({
+						uuid,
+						appId: 1,
+						gitHubAppName: "my awesome app",
+						gitHubBaseUrl,
+						gitHubClientId: "lvl.1n23j12389wnde",
+						gitHubClientSecret: "secret",
+						webhookSecret: "anewsecret",
+						privateKey: "privatekeyversion2",
+						installationId
+					});
+
+					const updatedApp = await GitHubServerApp.findForUuid(uuid);
+					expect(updatedApp && updatedApp.uuid).toEqual(gitHubServerApp.uuid);
+					expect(updatedApp && updatedApp.gitHubClientId).toEqual("lvl.1n23j12389wnde");
+					expect(updatedApp && updatedApp.webhookSecret).toEqual("anewsecret");
+					expect(updatedApp && updatedApp.privateKey).toEqual("privatekeyversion2");
+				});
+
+				it("should not update GitHub app when uuid is not found", async () => {
+					const mismatchedUUID = "c97806fc-c433-4ad5-b569-bf5191590ba9";
+
+					await GitHubServerApp.updateGitHubAppByUUID({
+						uuid: mismatchedUUID,
+						appId: 1,
+						gitHubAppName: "my awesome app",
+						gitHubBaseUrl,
+						gitHubClientId: "lvl.1n23j12389wnde",
+						gitHubClientSecret: "secret",
+						webhookSecret: "anewsecret",
+						privateKey: "privatekeyversion2",
+						installationId
+					});
+
+					const updatedApp = await GitHubServerApp.findForUuid(mismatchedUUID);
+					expect(updatedApp && updatedApp.uuid).not.toEqual(gitHubServerApp.uuid);
+					expect(updatedApp && updatedApp.gitHubClientId).not.toEqual("lvl.1n23j12389wnde");
+					expect(updatedApp && updatedApp.webhookSecret).not.toEqual("anewsecret");
+					expect(updatedApp && updatedApp.privateKey).not.toEqual("privatekeyversion2");
+				});
+
+				it("should only update values changed and leave other values as is", async () => {
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					await GitHubServerApp.updateGitHubAppByUUID({ uuid, appId: 1, gitHubBaseUrl, webhookSecret: "updatedSecret" });
```

Comment:
We shouldn't be able to submit a form with missing values as the frontend validation will fail. But added this test as proof that values in the db remain unaffected if data is missing from the payload.